core/support.go

```diff
@@ -55,29 +142,38 @@ func NewSupportClient() *SupportClientImpl {
 	}
 }
 
+// RequestServiceLimitsRefreshLoop ...
 func (client *SupportClientImpl) RequestServiceLimitsRefreshLoop() {
-	input := &support.RefreshTrustedAdvisorCheckInput{
-		CheckId: aws.String(checkId),
-	}
+	var (
+		waitMs        int64
+		refreshStatus string
+	)
 
 	for {
-		glog.Infof("Refreshing Trusted Advisor check '%s'...", checkId)
-		output, err := client.SupportClient.RefreshTrustedAdvisorCheck(input)
-		if err != nil {
-			glog.Errorf("Error when requesting status refresh: %v", err)
-			continue
+		for _, checkID := range checkIDs {
+			input := &support.RefreshTrustedAdvisorCheckInput{
+				CheckId: aws.String(checkID),
+			}
+			glog.Infof("Refreshing Trusted Advisor check '%s'...", checkID)
+			output, err := client.SupportClient.RefreshTrustedAdvisorCheck(input)
+			if err != nil {
+				glog.Errorf("Error when requesting status refresh: %v", err)
+				continue
+			}
+
+			waitMs = *output.Status.MillisUntilNextRefreshable
```

Comment:
Here, we overwrite `waitMs` according to the current check's `MillisUntilNextRefreshable`, and use the last one outside the loop to sleep.

I don't know if there are different refresh periods for different checks. If there are, maybe we can change the logic to keep the minimum `MillisUntilNextRefreshable` across all checks, and sleep for that period. Not very efficient (since we would try to refresh some checks that haven't reached the next update window) but simple to implement.

Another (more complex, even though more accurate) option would be to create a Goroutine for each check, and sleep for the correct amount of time in each check.

My suggestion is to go with the first option.